README.md

```diff
@@ -20,5 +20,30 @@ Scratch Blocks brings together two different programming "grammars" that the Scr
 
+## Internationalization and translations
+The scratch-blocks project aims to implement internationalization (i18n) for a wide variety of languages.
+
+We use the [Transifex](https://www.transifex.com/llk/public/) service to facilitate translation of Scratch Blocks by an international network of volunteers.
+
+### Source file
+* `msg/messages.js` is the source file for English strings. *Add new strings to this file directly.*
+
+### Generated files
+* `msg/json/en.json` is generated from `messages.js`. `en.json` is used to upload to Transifex (which requires JSON, not raw js). *Developers should not edit this file directly.*
+* `msg/js/en.js` is also generated from `messages.js`. `en.js` is used only by Blockly, and not used in Scratch. *Developers should not edit this file directly.*
+
+### Production file
+* `msg/scratch_msgs.js` is generated from Transifex data, and used by Scratch Blocks code when running. It is structured as a single object with a key-value pair for each supported locale; the value contains the translations for each string. *See note in "Development workflow" below about when to edit this file.*
+
+### Translation scripts
+* `npm run translate:sync:translations` downloads the latest translations from Transifex, and updates `msg/scratch_msgs.js` with them. It validates messages, so it can fail if there are errors in translations, such as invalid syntax. However, note that only approved users have privileges to run this script.
+
+### Development workflow
+Currently the source file `messages.js` and the translation file `scratch_msgs.js` must be kept in sync, or the test:messages task will fail.
+
+We do this daily using a scheduled cron job, but if you wish to develop locally with new strings defined, you will have to manually add your new strings to `scratch_msgs.js`. You can simply use the same English string; it will eventually be replaced with a proper translation.
+
+*If you submit a pull request which adds new strings to `messages.js`, your pull request should also include adding the strings to each locale in `scratch_msgs.js`.*
+
 ## Donate
-We provide [Scratch](https://scratch.mit.edu) free of charge, and want to keep it that way! Please consider making a [donation](https://secure.donationpay.org/scratchfoundation/) to support our continued engineering, design, community, and resource development efforts. Donations of any size are appreciated. Thank you!
\ No newline at end of file
+We provide [Scratch](https://scratch.mit.edu) free of charge, and want to keep it that way! Please consider making a [donation](https://secure.donationpay.org/scratchfoundation/) to support our continued engineering, design, community, and resource development efforts. Donations of any size are appreciated. Thank you!
```

Comment:
@chrisgarrity I'm not sure why Github thought this was changed -- there is no change to the text, but maybe there's a change to the whitespace?